core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -199,7 +229,9 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
     this.synchronized {
       if (!stoppedProcessingRequests)
         stopProcessingRequests()
-      requestChannel.shutdown()
+      dataRequestChannel.shutdown()
+      if (controlPlaneRequestChannel != null )
+        controlPlaneRequestChannel.shutdown()
```

Comment:
with the above suggestion : 
```suggestion
        controlPlaneRequestChannelOpt.map(_.shutdown())
```